docs/components/image/ImageFigure.tsx

```diff
@@ -0,0 +1,43 @@
+import React from "react";
+import Image from "next/future/image";
+
+type ImageProps = Parameters<typeof Image>[0];
+
+export type ImageFigureProps = ImageProps & {
+  caption?: string;
+  margin?: number;
+  captionSpacing?: number;
+  shadow?: boolean;
+  borderRadius?: boolean;
+};
+
+export function ImageFigure(props: ImageFigureProps): React.ReactNode {
+  const {
+    caption,
+    margin = 40,
+    captionSpacing = null,
+    shadow = false,
+    borderRadius = false,
+    ...rest
+  } = props;
+
+  return (
+    <figure className="block text-center" style={{ margin: `${margin}px 0` }}>
+      <div
+        className="relative inline-block w-full max-w-full overflow-hidden border-box"
+        style={{ fontSize: 0 }}
```

Comment:
Nitpick:

Could use `text-[0px]` in the regular classname unless you need this here for specificity

Reply:
in future: make the suggested commit / line change in github